apis/database/v1beta1/nosql_types.go

```diff
@@ -0,0 +1,290 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/azure-sdk-for-go/services/cosmos-db/mgmt/2015-04-08/documentdb"
+	"github.com/Azure/go-autorest/autorest/to"
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/pkg/errors"
+
+	apisv1alpha3 "github.com/crossplane/provider-azure/apis/v1alpha3"
+)
+
+// Error strings
+const (
+	errResourceIsNotValidNoSQL = "the managed resource is not CosmosDB"
+)
+
+// ResourceGroupNameReferencerForNoSQLAccount is an attribute referencer that
+// resolves the name of a the ResourceGroup.
+type ResourceGroupNameReferencerForNoSQLAccount struct {
+	apisv1alpha3.ResourceGroupNameReferencer `json:",inline"`
+}
+
+// ToDatabaseAccountCreate from DatabaseAccountSpec
+func ToDatabaseAccountCreate(s *NoSQLAccountSpec) documentdb.DatabaseAccountCreateUpdateParameters {
+	if s == nil {
+		return documentdb.DatabaseAccountCreateUpdateParameters{}
+	}
+
+	//DatabaseAccountProperties
+	return documentdb.DatabaseAccountCreateUpdateParameters{
+		Kind:                                  s.ForProvider.Kind,
+		Location:                              toStringPtr(s.ForProvider.Location),
+		Tags:                                  *to.StringMapPtr(s.ForProvider.Tags),
+		DatabaseAccountCreateUpdateProperties: toDatabaseProperties(&s.ForProvider.Properties),
+	}
+}
+
+// ToDatabaseAccountUpdate from DatabaseAccountSpec
+func ToDatabaseAccountUpdate(s *NoSQLAccountSpec) documentdb.DatabaseAccountCreateUpdateParameters {
+	if s == nil {
+		return documentdb.DatabaseAccountCreateUpdateParameters{}
+	}
+
+	return documentdb.DatabaseAccountCreateUpdateParameters{
+		Kind:                                  s.ForProvider.Kind,
+		Location:                              toStringPtr(s.ForProvider.Location),
+		Tags:                                  *to.StringMapPtr(s.ForProvider.Tags),
+		DatabaseAccountCreateUpdateProperties: toDatabaseProperties(&s.ForProvider.Properties),
+	}
+}
+
+// Assign assigns the retrieved group name to the managed resource
+func (v *ResourceGroupNameReferencerForNoSQLAccount) Assign(res resource.CanReference, value string) error {
+	switch sql := res.(type) {
+	case *CosmosDBAccount:
+		sql.Spec.ForProvider.ResourceGroupName = value
+	default:
+		return errors.New(errResourceIsNotValidNoSQL)
+	}
+	return nil
+}
+
+// +kubebuilder:object:root=true
+
+// A CosmosDBAccount is a managed resource that represents an Azure CosmosDB account with CosmosDB API.
+// +kubebuilder:printcolumn:name="STATUS",type="string",JSONPath=".status.bindingPhase"
```

Comment:
status and state are not displayed. The jsonPath does not work. It is the same as used for MySQL and Postgres, so it seems to be a generalized problem in this provider's code. I will fix it in a separate PR.

Reply:
`bindingPhase` will likely not appear unless you actually claim the resource. Have you verified that it is still blank when you claim? `status.state` below is not a field in `runtimev1alpha1.ResourceSpec` so I would not expect it to be exposed. I'll take a look at `MySQL` and `Postgres` to see if they are broken :+1: